src/docker/install.ts

```diff
@@ -92,6 +93,26 @@ export class Install {
       extractFolder = path.join(extractFolder, 'docker');
     }
     core.debug(`docker.Install.download extractFolder: ${extractFolder}`);
+    return extractFolder;
+  }
+
+  public async download(): Promise<string> {
+    let extractFolder: string;
+
+    core.info(`Downloading Docker ${this.version} from ${this.channel}`);
+
+    this._version = this.version;
+    if (this.version == 'master') {
```

Comment:
I think it would be better to rely on csv values so we can either choose archive or image format:

* `v24.0.9` similar to `type=archive,version=v24.0.9`
* `type=image,tag=master`
* `type=image,tag=24.0.9`

WDYT?

Reply:
But that would be smth to handle in `setup-docker` action directly.

Reply:
Do we want to have another input variable for cli version alone?
Like:

```
version: type=image,tag=master # and optional repository=moby/moby-bin-someotherepo 
cli-version: type=archive,version=v20.10.25 # if not specified, will default to tag from `version` an
```

or just sth like
```
version: |
  component=engine,type=image,tag=master
  component=cli,type=archive,version=v20.10.25
```

Reply:
Or, if we want to go in the direction of: https://github.com/crazy-max/ghaction-setup-docker/issues/81

Then maybe we should have a separate `image` input:
```
    images: |
      moby/moby-bin:master
      dockereng/cli-bin:master
```
that would be mutually exclusive with `version`?

Reply:
> Do we want to have another input variable for cli version alone?

I think we should keep both under the same terminology instead of being separated. Don't think we want to allow custom registry or image repository imo.

> Or, if we want to go in the direction of: [crazy-max/ghaction-setup-docker#81](https://github.com/crazy-max/ghaction-setup-docker/issues/81)

I think csv values like https://github.com/docker/actions-toolkit/pull/438#discussion_r1746917839 are easier to maintain so we don't introduce a new input and keep backward compat with current `version` input. WDYT?

Reply:
My main concern here was the ability to install a different engine and cli version. I think it's fine to leave the image repository hardcoded, but it could be useful to be able to install different versions of CLI and Engine.

Reply:
> but it could be useful to be able to install different versions of CLI and Engine.

Ok if there is this use case then what do you think of:

* `v24.0.9` > `type=archive,version=v24.0.9` or `type=archive,engine_version=v24.0.9,cli_version=v24.0.9` if you want specific engine/cli version

Same for images:

* `type=image,tag=24.0.9` > `type=image,engine_tag=24.0.9,cli_tag=24.0.9`

Reply:
SGTM, although it's not an urgent use-case so we can handle that in a follow up.

Reply:
Updated to support the CSV syntax.
Example run: https://github.com/docker/moby-private/actions/runs/10773693004/job/29874033854?pr=9#step:2:336